bundles/http/Citrix/Covid-19-Self-Certify/script.js

```diff
@@ -0,0 +1,246 @@
+const uuid = library.load('uuid')
+integration.define({
+    synchronizations: [
+        {
+            name: "syncPodio",
+            fullSyncFunction: syncApp
+        }
+    ],
+    model: {
+        tables: [
+            {
+                name: "items",
+                columns: [
+                    { name: "employeeName", type: "STRING", length: 255 },
+                    { name: "email", type: "STRING", length: 255, primaryKey: true },
+                    { name: "confirmation", type: "STRING", length: 255 },
+                    { name: "createdOn", type: "DATETIME" },
+                    { name: "app_item_id", type: "INTEGER" },
+                    { name: "region", type: "STRING", length: 255 },
+                    { name: "country", type: "STRING", length: 255 },
+                ]
+            },
+            {
+                name: "cities",
+                columns: [
+                    { name: "uniqueId", type: "STRING", length: 255, primaryKey: true },
+                    { name: "countryName", type: "STRING", length: 255 },
+                    { name: "regionName", type: "STRING", length: 255 },
+                ]
+            },
+            {
+                name: "countryConditions",
+                columns: [
+                    { name: "uniqueId", type: "STRING", length: 255, primaryKey: true },
+                    { name: "guidelines", type: "STRING", length: 3000 },
+                    { name: "region", type: "STRING", length: 255 },
+                    { name: "image", type: "STRING", length: 255 }
+                ]
+            }
+        ]
+    },
+    integrationParameters: [
+        { name: "covidAppId", type: "INTEGER", label: "Covid Self Certification App Id", required: true },
+        { name: "viewId", type: "INTEGER", label: "Covid Self Certification View Id", required: true },
+        { name: "countryDataAppId", type: "INTEGER", label: "Country Data App Id", required: true },
+    ],
+    actions: [
+        {
+            name: "covid_self_certify",
+            parameters: [
+                { name: "email", type: "STRING", required: true },
+                { name: "status", type: "INTEGER", required: true },
+                { name: "country", type: "STRING", required: true },
+                { name: "city", type: "STRING", required: true },
+                { name: "appId", type: "STRING" }
+            ],
+            function: covidSelfCertify
+        }
+    ]
+})
+
+async function validateResponse(response){
+    let responseBody, errorMessage, errorStatus
+    if(response.headers.map['content-type'].includes("application/json")){
+        const jsonBody = await response.json()
+        if(!response.ok){
+            responseBody = jsonBody
+            errorMessage = jsonBody.error_description
+            errorStatus = true
+        }else{
+            responseBody = jsonBody
+            errorStatus = false
+        }
+    }else{
+        const body = await response.text()
```

Comment:
bug, response body is a buffer which is already processed and will throw an error when accessed again.

Reply:
Changes Done